src/mlpack/methods/ann/layer/linear_impl.hpp

```diff
@@ -94,11 +94,11 @@ LinearType<MatType, RegularizerType>::operator=(
 
 template<typename MatType, typename RegularizerType>
 void LinearType<MatType, RegularizerType>::SetWeights(
-    typename MatType::elem_type* weightsPtr)
+    const MatType& originalWeights)
 {
-  MakeAlias(weights, weightsPtr, outSize * inSize + outSize, 1);
-  MakeAlias(weight, weightsPtr, outSize, inSize);
-  MakeAlias(bias, weightsPtr + weight.n_elem, outSize, 1);
+  MakeAlias(weights, originalWeights, outSize * inSize + outSize, 1);
+  MakeAlias(weight, originalWeights, outSize, inSize);
+  MakeAlias(bias, originalWeights.at(weight.n_elem), outSize, 1);
```

Comment:
I don't think this will compile; I think you will need to adapt `MakeAlias()` so that it takes an `offset` parameter.